rfc6962-bis.xml

```diff
@@ -1319,12 +1319,15 @@ but it is expected there will be a variety.
             In addition to normal validation of the server certificate and its chain, TLS clients SHOULD validate each received SCT for which they have the corresponding log's metadata. To validate an SCT, a TLS client computes the signature input from the SCT data and the corresponding certificate, and then verifies the signature using the corresponding log's public key. TLS clients MUST NOT consider valid any SCT whose timestamp is in the future.
           </t>
           <t>
-            Since each received SCT may be valid either for the server certificate itself or for a <xref target="name_constrained">suitable name-constrained intermediate</xref>, TLS clients MAY need to use trial and error to determine which certificate in the chain is the certificate to which a received SCT corresponds.
+            Before considering any SCT to be invalid, the TLS client MUST attempt to validate it against the server certificate and against each of the zero or more <xref target="name_constrained">suitable name-constrained intermediates</xref> in the chain. These certificates may be evaluated in the order they were sent by the TLS server, or, indeed, in any order.
```

Comment:
I think the final sentence ("These certificates may be evaluated in the order they were sent by the TLS server, or, indeed, in any order.") should be removed:
(1) it may not be feasible to require that TLS clients evaluate SCTs on the chain received, rather than the chain built by the client, as different clients have different heuristics for building chains.
(2) What if a server sends a name-constrained intermediate that is not even a part of the chain, but sends a valid SCT for it? I think that in that case the client should not consider the leaf to have a valid SCT covering its chain.

To sum it up, I suggest we not get involved with the way clients build their chains and simply say that the client should attempt validating the SCT against each of the suitable name-constrained intermediates in it.